ecl/eclcc/eclcc.cpp

```diff
@@ -126,16 +128,54 @@ void __cdecl IntHandler(int)
     exit(2);
 }
 
-#include <signal.h> // for signal()
-
-MODULE_INIT(INIT_PRIORITY_STANDARD)
+static void installSignalHandlers()
 {
     signal(SIGINT, IntHandler);
-    return true;
+}
+
+static void restoreSignalHandlers()
+{
 }
 
 #else
 
+//If eclcc is terminated while a git fetch operation is in process then there are situations where orphaned
+//lock files are left in the directory.  To avoid this, prevent termination until the git fetch is complete.
+static std::atomic<unsigned> terminateRequests = 0;
+static void sighandler(int signum, siginfo_t *info, void *extra)
+{
+    if (!checkAbortGitFetch() && (++terminateRequests < 2))
+    {
+        //Delaying termination while fetching from git.  Terminate 2 times (or use kill -9) to force closure.
+        //Cannot output any logging here because that is not a safe function to call from an interupt handler
+    }
+    else
+    {
+        enableMemLeakChecking(false);
+        _exit(2);
+    }
+}
+
+static void installSignalHandlers()
+{
+    struct sigaction act;
+    act.sa_sigaction = &sighandler;
+    sigemptyset(&act.sa_mask);
+    act.sa_flags = SA_SIGINFO; // set so sa_sigaction field is used
```

Comment:
Could add SA_RESTART to sa_flags maybe to help any syscall that was interrupted in doMain().